core/src/apps/lisk/sign_tx.py

```diff
@@ -0,0 +1,134 @@
+from trezor import wire
+from trezor.crypto.curve import ed25519
+from trezor.enums import LiskTransactionModuleID, MessageType
+from trezor.messages import LiskSignedTx
+from trezor.protobuf import dump_message_buffer, load_message_buffer
+
+from apps.common import paths
+from apps.common.keychain import auto_keychain
+
+from . import layout
+from .helpers import get_lisk32_from_address_hash
+
+
+@auto_keychain(__name__)
+async def sign_tx(ctx, msg, keychain):
+    await paths.validate_path(ctx, keychain, msg.address_n)
+
+    # decode the buffer into LiskTransaction
+    tx = load_message_buffer(msg.transaction, MessageType.LiskTransaction)
+
+    pubkey, seckey = _get_keys(keychain, msg)
+
+    # check if the sender_public_key is the same as the derived one
+    if not _check_sender_publick_key(tx, pubkey):
+        raise wire.DataError("The transaction has invalid sender public key")
+
+    # decode the asset based on module_id and asset_id
+    asset = _decode_tx_asset(tx)
+    # Reset asset with the dump (so we are sure we are signing only what we expect)
+    tx.asset = dump_message_buffer(asset)
+
+    try:
+        await _require_confirm_by_type(ctx, tx, asset)
+    except AttributeError:
+        raise wire.DataError("The transaction has invalid asset field")
+
+    amount = _get_tx_amount(tx, asset)
+    await layout.require_confirm_fee(ctx, amount, tx.fee)
+
+    txbytes = dump_message_buffer(tx)
```

Comment:
please add check that `txbytes == msg.transaction` and raise a `DataError` if there is mismatch

Reply:
Good idea yes. will do